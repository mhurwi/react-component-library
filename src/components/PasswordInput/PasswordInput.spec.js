import React from "react";
import renderer from "react-test-renderer";
import PasswordInput from "./PasswordInput";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

test("toggles input type when show/hide password clicked", () => {
  const wrapper = shallow(
    <PasswordInput
      htmlId="test"
      name="test"
      onChange={() => {}}
      value=""
      showVisibilityToggle
    />
  );

  expect(wrapper.find({ type: "password" })).toHaveLength(1);
  expect(wrapper.find({ type: "text" })).toHaveLength(0);

  wrapper.find("a").simulate("click");
  expect(wrapper.find({ type: "password" })).toHaveLength(0);
  expect(wrapper.find({ type: "text" })).toHaveLength(1);
});

test("hides password quality by default", () => {
  const tree = renderer
    .create(
      <PasswordInput
        htmlId="test"
        name="test"
        onChange={() => {}}
        value="AaSsDdFf123#!"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
