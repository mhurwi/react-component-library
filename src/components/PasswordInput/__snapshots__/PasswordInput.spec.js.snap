// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides password quality by default 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label
    htmlFor="test"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Password
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    id="test"
    name="test"
    onChange={[Function]}
    placeholder={undefined}
    style={undefined}
    type="password"
    value="AaSsDdFf123#!"
  />
</div>
`;
